test/acceptance/libp2p/host/protocol/client_test_session.cpp

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "acceptance/libp2p/host/protocol/client_test_session.hpp"
+
+#include <boost/assert.hpp>
+#include "libp2p/crypto/random_generator/boost_generator.hpp"
+
+namespace libp2p::protocol {
+
+  ClientTestSession::ClientTestSession(
+      std::shared_ptr<connection::Stream> stream, size_t client_number,
+      size_t ping_times)
+      : stream_(std::move(stream)),
+        client_number_{client_number},
+        messages_left_{ping_times} {
+    BOOST_ASSERT(stream_ != nullptr);
+    random_generator_ =
+        std::make_shared<crypto::random::BoostRandomGenerator>();
+  }
+
+  void ClientTestSession::handle(Callback cb) {
+    write(std::move(cb));
+  }
+
+  void ClientTestSession::write(Callback cb) {
+    if (messages_left_ == 0) {
+      return;
+    }
+
+    --messages_left_;
+
+    if (stream_->isClosedForWrite()) {
+      return;
```

Comment:
I guess stream should not be closed for Write in tests, unless you explicitly test stream close case

Reply:
added check